fv3core/_config.py

```diff
@@ -73,6 +73,93 @@
 DEFAULT_BOOL = False
 
 
+@dataclasses.dataclass(frozen=True)
+class RiemannConfig:
```

Comment:
Config is pretty generic. It might be helpful for users to explicitly call it NamelistConfig, or just Namelist (e.g. RiemannNamelist), so it is clear what kind of a configuration it is and where it typically comes from.

Reply:
Maybe, I wanted to separate the word Namelist from Config though to create a separation of "this class corresponds directly to the Fortran namelist, uses the same variable names, and has a similar structure" from "these are internal new configuration classes where we might use different names, consolidate multiple namelist settings, or change types of settings (e.g. integers to descriptive strings)" with functions that map between the two. We can chat about it.

Reply:
Also consider that an analysis user might want to use the Riemann object directly outside of a model, and they'd be making a RiemannConfig in this case. The arguments could be hard-coded then, as opposed to coming from any sort of file. It being "config" is really all we know for sure about this data.

Reply:
I can see the intent here, I worry it makes it unnecessarily confusing for a new user. Something being a Namelist object doesn't necessarily mean it has to come from a file on disk, but indicates it is part of the model algorithm options configuration. The Riemann is one of the few modules that could have the potential to get used in other contexts, most are highly integrated and rely on other parts of this model.  If someone is in the riemann solver code, it should be obvious to them where the configuration settings came from and how to change them. If a configuration object combines settings set in a variety of sources, that is not super user-friendly for them understanding the model. We already have quite a handful of varieties of 'configurations' defined and set in different places in the model, and I think 'Config' is not specific enough. 
I also think we shouldn't be internally combining and changing these options, they are already a lot to understand, and adding additional mappings and interpretation layers will make it difficult for anyone new to work on and improve this code. It will also make it easier to make configuration mistakes. Especially since we haven't ported all existing options, changing the way objects are configured will make doing that work harder than it needs to be. We should use the settings as provided by the Fortran model and keep it as straightforward as possible at this stage, in my opinion. 
I'm a fan of using configuration classes as you have laid out though, with classes specific to a particular part of the model. That seems useful.